\section{Biological Introduction}

\subsection{Tree of Life and the Theory of Life}
The tree of life is both the organizing structure of life sciences (like it's predecessor- Linnean classification) and is a representation of the theory of life - evolution.  On a short time-scale it is a picture of biological reproduction - or unfaithful cloning - a pedigree.  On long time-scales, the tree is a representation of the process of species evolution, which in many ways can be thought of as a pedigree of species, where each node represents a population of a particular specie, and descent down the branches represents time, and division of a node into two nodes (reproduction in a standard pedigree - parents having children..) represents speciation events from a common ancestor (reproduction at a population level on geological time-scales).
\subsubsection{Comparative Anatomy and Physiology}
Previously,in the eighteenth century, Linnean by brute force clustering techniques organized life according to anatomical simililarites, leading to a classification system for life. This organization was not just a Victorian database that organized collections of living objects by comparative anatomy; the organization explained why the objects were different or why they were similar because comparative anatomy begs the question of 'function' of physiology, and physiology is a theory of life.   In this sense, Linnean systematic organization of life was a simple theory of life, in that it did explain life, as does all physiology, because it explained the purpose (function) of each anatomical feature\footnote{In biology the 'function' of a trait or anatomical structure is what the trait does or accomplishes, hence the word 'purpose' seems to be a synonym to function.  However, this may be confusing seeing that evolution has no 'purpose', hence 'purpose' should not be interpreted as if the population or lineage that evolved the trait had foresight and intended its creation.}.  For example, the purpose of a leg is locomotion, of a jaw is to bite, of a root is to stabilize a standing plant (among other functions), of a circulatory system (heart) is to circulate nutrients throughout the organism, of a stamen is for sexual reproduction etc.; and Linneaus knew these trivial relationships between structure and function, which undoubtedly helped in his grouping of organisms.  Interpreting the organization of life through a theory of structure and function is very powerful, as a theory intentional simplifies complex patterns so that they can be understood and comprehended.  Hence the theory of 'structure and function' has one of the most important aspects of biological theory, and that is to simplify life in a way that we can comprehend its rich diversity.  It also has an obvious predictive power, for example if an animal loses its legs you can predict that it will lose locomotion.  Although many of Linnean groups are based on reproductive anatomical features, a 'structure and function' theory is very short-sighted in terms of how life reproduces, as it can only explain how to maintain existing population of species (by having the reproductive structures do what they do).  What's clearly missing is the origin of life, how to make life from scratch; and the origin of all the different kinds or species (i.e. specie meaning a group that can form viable offspring).

The theory of evolution by Charles Darwin, which I'll summarize in two pieces 'descent' and 'modification' added  more structure to the theory of life.  Darwin recognized (hypothesized) that common anatomical structures were due to common 'descent', indicating common features need not be derived from scratch using different ingredients, the entire structure was passed on during reproduction.  He also recognized that different structures between two groups of animals was due to 'modification' from a common ancestor between the two groups of animals.

\subsubsection{Classical evolution}

Darwin's theory united life through one common lineage.  However the biological mechanisms of how to produce life from molecules (e.g. how are completely novel and complex features derived in the first place), or to show that one specie of organism could be related to another (through the common ancestor) were not sufficiently convincing to some people such that it be considered \textit{the} theory of life.  That would require the elucidation of two parts of molecular biology.  

Before the advent of molecular biology, from a reductionist point of view, the proof (evidence) of the evolutionary tree of multicellular organisms can be thought of as a hiearchy of conservation of features or modules.  The importance of conservation can not be stated enough, as Darwin's central tenant is 'descent', meaning common features between our ancestors and ourselves are due to conservation from descent.  At the highest level of modularity (and by far the most useful for a big picture resolution of the tree of life for multicellular orgaisms) is the anatomical structures (e.g. leg, eye, heart, nervous system).  Comparative embryology has elucidated that all these anatomical structures from organs to gross features like a leg (containing multiple organs, like skin) are derived from possibly just three tissue types (germ layers).  For example, the heart is derived from the mesoderm tissue, and all multicellular organisms without a mesoderm in their embryoinic stage do not develop a heart.  These three tissue types are composed of largely undifferentiated cells and during development they work together and sometimes work independently to fate the array of different cell types that make up the anatomical features that make up multicellular organisms (such as epidermal cells, hemapoatic cells, blood cells).  Hence, before molecular biology (before the 'gene' picture), there was at least three basic conserved traits that could be used for evidence on resolving tree branching in multicellular organisms (anatomical features, germ layers, cell types).
    
\subsubsection{Modern Synthesis of evolution}

\textbf{Molecular genetics}, the first contribution from molecular biology, would show animals posses genes that are contained in chromatin and that those genes were passed on from parents to children during reproduction, this culminated in the 'modern synthesis' of evolutionary biology in the early 1900s.  This would gain further support by R. Franklin's crystallization of a chunk of chromatin, DNA.  DNA was found to be a polymer strand that would complement with another self assembling strand, which Francis and Crick saw could serve as a 'copy' for a replicating cell's progeny cell.  The 'modern synthesis' is largely about the molecular dynamics of populations, 'population genetics', a population of organism's genomes (the frequency of genotypes) and how those change in time (in units of generations), and how they can be influenced by natural selection, mutation, gene flow, migration and isolation (E. Mayer type speciation).  Most significantly, the gene centered picture that had arisen in evolution now had given a fourth feature or module that was conserved, the gene that encodes for a protein. 

\subsubsection{The generalization of the theory of evolution}

The second part of molecular biology that supports macroevolution, and possibly an extension of the 'modern synthesis', is \textbf{developmental genetics}. Developmental genetics would show that master genes ( transcription factors) when their regulatory targets or binding sites evolved, then whole developmental networks (i.e. the genes necessary to build an anatomical structure) could be redeployed to a different position of a developing body (e.g. the position that expressed the activator master gene), or lead to modifications of current body parts\footnote{As suggested by the phrase, 'developmental genetics', this would seem to be a subdiscipline of 'molecular genetics' and hence not an extension of the modern sythesis, but rather a refining.  However, as I will discuss, the modern synthesis, was a gene centered theory.  It was about gene's that encode for proteins.  Developmental genetics is a gene 'regulatory' theory, it's all about the parts of the genome that turn traditional genes and and off; and gene regulatory elements are not really apart of the modern idea of a gene.  A gene encodes for a protein.  A genetic regulatory element encodes for something altogether different, and not a part of the modern synthesis.}.  

Developmental genetics, many believe, is \textit{the} field of study that shows at a molecular level how whole anatomical features evolve, and how the diversity of anatomical features and their arrangenment (body plans) has evolved.  This is because the detailed mechanisms that are needed to make Darwinian theory convincing at an anatomical scale (sometimes crudely called 'macroevolution') are observed in the field of developmental genetics through powerful experiments such as transplantation experiments, and gain and loss of functions experiments, along with the powerful tools of genetics and microscopy techniques.  In a sense developmental genetics, elucidates the 'Gene Reulatory Networks' that form the causal coarse grained molecular basis of self assembling anatomical features. 
\subsection{Development}

\subsubsection{The origin of multicellularity; the evolution \textit{of} development}
About a billion or two years ago, single celled algae started to cooperate by developing cell cell interactions, forming the first eukaryotic multicellular  organisms, like algal mats.  Such constructs eventually lead to the important innovation of multicellular organisms that is found in animals (and other kingdoms): sexual reproduction through meiosis and fertilization.  Like endosymbiosis, which is possibly the origin of eukaryoutes from bacteria, where two cells would merge and partner to share their particular genes and hence traits (which may be different traits), in sexual reproduction cells not only merge, or fertilize (merging of two gametes into a zygote), but they first go through a phase which is known as meiosis, which is significantly different from the results of endosymbiosis (which replicate through mitosis of each fused component) significantly different due to the 'recombination' of traits, or crossing over, leading to great diversity in progeny, thereby leading to faster evolution (by Fisher's Fundamental Theorem) and preventing Muller's ratchet (by creating gamete's free of detrimental mutations).  Furthermore, the fusion of two genomes (e.g. the two gamete genomes) is a form of 'gene duplication', which is a source in 'gene families', where the copied gene leads to diversity in function of family of genes\footnote{Two bacterial cells could fuse leading to a n -> 2n genotype, similar to diploidy, and allowing for greater genomic diversity as one of the duplicate copies of the gene are now possibly free to serve a new function.  However, this genome duplication event is distinct from sexual reproduction (and hence is not diploidy, in my opinion), as the new bacteria with 2n genes, can be said to now have simply n' genes (a haploid with n' genes), when the n' genes are replicated (through mitosis), the progeny are clones (if mutation rate is slow enough), which distinguishes it from sex, in sexual reproduction the progeny are not clones due to recombination (assuming the mutation rate is high enough that there exists some genetic polymorphisms in the population, such that the cross over pieces are not identical).}.  

These early sexual reproducing algea are the origin of 'eukaryotic early development' (which I define as eukaryotic cellular interactions that lead to a multicellular stucture, such as an algal mat (a plain of cells), or a 'blastula' (a ball of cells)\cite{pmid7579526}).  They are modern representatives, living fossils, of the evolution of multicellularity.  

In the diverse domain of eukaryote the most famous groups are multicellular organisms, due to their gross anatomical features, in these multicellular organisms there is a remarkable common or conserved early development, yet there is also marked differences suggesting that multicellularity has independently evolved in plants and animals and fungus.  

Examples of multicellularity precursors to multicellularity can be even seen in bacteria in (quorom sensing) and the primitive fungus yeast through their form of sex using 'mating types' (analogs of males and females)\cite{wolpert}. An array of protists (eukaryotes that are not animals, plants, or fungus) show multicellularity, notably volvox and slime molds.  All these 'primitive' organisms are good starting points for the study of multicellularity.  However 'development' in academia, to a large degree, focuses on multi-cell types (a skin cell 'functions' differently than a germ cell and functions different than a stomach cell (digestive cell that can 'eat' absorbed surroundings ).  That is the division of labor through specialized cells, which is not seen in many primitive organisms, rather these simple organisms are displaying 'colonies', which are the aggregates of cells due to mitosis resulting in progeny cells being proximal to one another (which is physical important in development, but it doesn't display the division of the aggregate into different cell types).

The starting point of 'development' is the innovation of meiosis, invented by the protists, and passed on to its progeny lineages of plants, animals, and fungus.  Hence, plants and animals don't derive sex for themselves from scratch, they were the benefactors of it from a protistan ancestor.  By the union of meiosis with fertilization (the opposite of meiosis, in a sense) the ability to always have an extra copy of a gene, and therefore the ability of a gene to evolve a new function becomes a stable component of life forms (regardless of whether some life forms display a 'haploid dominant' life form, like ferns, where the plant one normally sees is the haploid, this is because most of the ferns \textit{life cycle} exist in the haploid stage (even when it's an 'adult')).


The great lineages of animals, fungus and plants all develop from the fertilized 'egg' (the fusion of the two gametes).  'Egg' means oocyte here, one of the gametes, while in development literature, egg may also mean the structure that encases a baby, like a chicken egg.  Initially in evolution, possibly, there was no distinction between gametes, egg and sperm, both gametes were equal, just haploid cells, like in yeasts (which don't go through meiosis, just fertilization).  Regardless of the origin of meiosis, we see here a clear case of different cell types and an abstract case of the division of labor at the cell level, the emergence of multi-cell types (which, again, is different than simply cell aggregation).  The division of cell types for the first multi-cell type organisms is speculative, but it seems the haploid cell's role in the life cycle of the first meiotic cells was to provide a means to generate unique progeny that were not clones of the parents (through recombination).   

Cell aggregation in the form of colonies or even in the form of complex structures (the rudiments of a body plan) such as seen in the protist volvox or the 'transitional form' between fungus and animals in the organism choanoflagellates is possibly evolved independently of meiosis (which i consider the origin of different cell types).  Cell aggregation is caused by gene products that connect cells to together such as cadherins, actin, hence one would suspect these genes would be conserved among the multicellular lineages, however a rather recent startling finding is that the great multicellular lineages of plants animals and fungus do not conserve these genes, and hence they have each evolved 'by scratch' (by convergence), and hence, in a sense, suggest that the origin of development can not simply be tied in to the origin and evolution of body plans (plants have body plans in the form of features like roots shoots and leaves, which can be laid out in many ways, are these related to the layout of animal body plans (like head thorax and tail?  Convergences or completely novel genes found in plants would suggest not. Parallel evolution of body plans between plants and animals and fungus does not mean we should not bother comparing their different life cycles and body plans or compare their anatomies and physiology.  For an excellant developmental comparative anatomy between plants and animals see Alberts.  The potential fact that these great lineages are all independent simply means that the powerful inferences (predictions) that can be made based on common ancestry are not valid (since their common ancestry is irrelevant if each one independently evolved their anatomical structures independently, hence the only thing constraining their diversity would (difference between plants and animals) would be basic physics).  However, their potential multicellular independence is not as independent as some may suggest, there is still the deep homolgy in that they all still use meiosis.  Albeit, this may be of limited help in understanding their origins seeing that plants have a 'cell wall' that is rigid, suggesting that the production of gametes, and the form of the gametes themselves is vastly different between plants and animals (and indeed pollen and how it 'grows' into the female stamen leading to the ovoum is very different than sperm fertilizing an egg in animals). 
/////////////////////////////
\subsubsection{Fly Development}

The coarse-grained in space and time molecular dynamics of how a fly is self assembled from a maternally laid egg, a single cell, is coarsely understood at the molecular level thanks to developmental genetics.  This does not explain or prove anatomical evolution, however if gain or lose or modification of a master genes (which encode transcription factors acting in early development) or of gene regulatory binding sites that interact with the master genes did result in modification or gain of loss of anatomical features, then this would convincingly explain the evolutionary mechanism of anatomical features\footnote{This type of huge leaps in modification (leaps) of an organism was observed in the fossil record by Jon Conway (among others), a palaeontologist, which he called the Cambrian Explosion, and is in contrast (contrast does not mean conflict or contradictory) to the ubiquitously accepted 'gradual accumulation' of beneficial mutations that results in adaptations within a particular genera of organisms (this was even known by plant and animal breeders before Darwin's theory).  The idea of saltation was called 'punctuated equilibrium' by Steven Jay Gould, where periods of gradual accumulation are punctuated by bursts of major anatomical features.}  

In the 1950s J. Monod working with bacteria discovered that transcription factors can influence gene expression by activating or repressing a gene, where the transcription factor itself was activated by environmental cues (such as sugar).  This would mark the start of the field of gene regulation, which would be the central theme in developmental genetics.  


In \textit{Drosophila} development it is known that after fertilization of the egg and after cleavage (mitosis without a G phase (cells don't grow)) within the blastocyst maternally laid transcription factors (such as Bicoid) zygotically regulate other transcription factors (such as gap genes (which are transcription factors) which in turn regulate pair-rule genes (which are transcription factors) which in turn regulate effector genes (like HOX genes some of which are transcription factors)) that ultimately feed in to signalling networks and paracrine factors that lead to the construction of anatomical features.  

The initial maternally laid transcription factors are not ubiquitously expressed throughout the egg chamber, rather, like a Fourier series, the first maternal transcription factor protein form a coarse pattern across the embryo, where the protein concentration is like a square wave of concentration as a function of space that forms a term in a Fourier series (i mean discrete summation of a few signals).  This transcription factor protein may act alone to activate a gap gene (like \textit{hunchback}) or may act with another maternally laid transcription factor whose pattern is also like a square wave (with a phase shift), thereby \textit{adding} another term to the series.  The addition of the two input signals results after a bit of time in an additional new pattern across the embryo in the form a gap gene's protein concentration.  Hence as time progresses more complex patterns appear as the gap gene's interact (primarily like \texit{addition} of waves in a Fourier series) resulting in patterns like a sine wave (where, again, the amplitude is the amount of protein at a particular time and the horizontal axis is a spatial axis of the embryo (the wave is in space not time), where the embryo's axis length is fixed in time).  Hence, after cleavage in development, the totipotent cells of the blastocyst are in a sense transforming into more specialized cell types, where the cell is defined by the amount of each specific gene product's protein concentration at a particular \textbf{position} of the embryo, where position is emphasized to stress that the concentration patterns are over space (the location of the embryo).  These differentiated cells will divide further and differentiate further as the embryo starts to take the form of an adult segmented fly.  The chain reaction of gene interactions that begins with the maternal transcription factors that activate other transcription factors, which in turn activate other factors, gives a molecular dynamics description of early development, and hence answers the question of how to build 'from scratch' a body plan (the arrangement of anatomical features).  

'From scratch' is misleading and ambiguous when discussing body plans of animals.  There are two issues the phrase needs to invoke, ontogeny (development) and phylogeny (evolution).  Disentangling the ambiguity we see first, how an adult arises from a single fertilized egg using the model system Drosophila.  Second, how did single celled organisms evolve as the diversity of all multicellular organisms.  It is in phylogeny and evolution that the question of how to evolve from scratch an animal is misleading.  Evolution does not build from scratch (i.e. independent evolution such as convergence or parallelisms), body plans are thought to derive through the reconstructions of a set of modules (devolopmental networks of genes) that each encode for anatomical features (hence all eyes, for example, are homologies at a developmental level through the master gene (transcription factor) \textit{pax} that binds to a set of genes that set a 'totipotent' or partially programmed cell to start to 'develop' the eye imaginal disc (the set of cells that form the basic outline of an eye, which when induced, will form an eye (at any part of a body that 'induction' occurs (even in the tail region if so induced).  

The idea of a totipotent cell, and cell specification, are necessary to explain development, and hence necessary component of an extension to the modern synthesis.  Cell differentiation and cell lineages I will briefly discuss below in the origin of multicellularity.  Of course, the initial body with many of the modules (urbiliteria, the ancestor of all animals) or its fungal and then algal ancestor, still needs to be explained as how one builds it from scratch (people are doing this with choanoflagellites and brine shrimp), but must people are satisfied with a hypothetical urbiliteria that can explain the diversity of animals by the evolution of gene regulatory networks.  Furthermore, the statement that anatomical features are modular at the genetic level is not obvious, as many believe complex features (anatomical structures) contain highly correlated genetic networks, such that any genetic mutation would be deleterious and probably lethal.  The discovery of the extent of modularity (or the extent of nonmodularity through 'induction' by signaling and paracrine factors) I will discuss briefly in the section below on 'modularity'.}.        
%\subsubsection{Comparative Embryology}

\subsubsection{Evolution of body plans in animals}

The diversity of the animal kingdom can not be explained by standard gene molecular evolution (evolution of gene's encoding proteins); the differences between proteins in animals is not sufficient to explain the diversity in animal phenotypes\cite{pmid1090005}.  It is thought there is about 20,000 proteins, and these proteins (for example, hemoglobin) are largely conserved across the entire kingdom.  The diversity is now thought, to a large extent, to be due to the way these proteins are used in different amounts and in different combinations in different cells and body parts of different multicellular organism\cite{King}.  To change the amount or create combinations of various proteins in a particular cell in an organism is accomplished by gene regulation, along with expansion of genome sizes (from the early bacterial and protist) through gene duplication allowing the roughly \textit{same} protein to adapt to a new niche in the organism.  Hence, in short, the diversity is due to the evolution of the elements of gene regulatory networks, where the transcription factor binding site is THE fundamental unit.
\subsection{The crowning jewel of evo-devo}
The work of Ed Lewis (a PhD student of Alfred Studevant) on body-transformative genes that were 'saltationary'\footnote{Saltation is the idea that within just one generation major evolutionary transformations could occur, possibly even speciation, taking the idea to its fictional extreme would be like an ape having a baby human, a 'hopeful monster', hence in one generation a speciation event occurred.} helped provide evidence and a theoretical framework (extending evolution theory) for the statement that diversity in the animal kingdom (in particular the \textit{segmented} insects) is due to evolution of gene reguatory networks.  The "Hox" genes were a group of genes that caused 'Homeotic transformations', which means a transformation in a body part of an animal, transformations that were known to some biologist such as Bateson as early as the late nineteenth century, who had catalogued these in various animal groups such as crabs and flys etc..  \textit{The} exemplary gene in the HOX group is 'bithorax', which was discovered in Thomas Hunt Morgan's fly lab in 1915, and a lineage of these mutants has been preserved since then. The gene is named after its mutation that lead to a mutant fly with two (bi) thoraxes (like an abdomen), which consequently doubles the number of wings of the fly (since the thorax is where the wing's developmental source (or pool of cells that each have the right genes turned on and off- a form of programming - leads to the developmental source of cells of the wing, these 'programmed' cells during early development are the so-called 'imaginal disc', and in this case, the wing imaginal disk\footnote{place a 'wing imaginal disc' in the location of the head, and you get wings growing out of the head, or place a 'leg imaginal disc' - the cells 'programmed' to build a leg where a wing is supposed to occur and you'll see leg's 'grow' or further develop where the wing's were supposed to be.} occurs, and hence is a homeotic transformation.  The Morgan lab didn't know the molecular genetic basis behind bithorax (what DNA sequence had changed (or possibly networks of DNA sequences) from the wild type fly to the mutant), but they did know the mutation was inheritable, and hence was genetic.  They also were able to 'map' the location of these genes (before they even knew genes were made of DNA) to locations on the chromosome due to a technique developed by Morgan's undergrad student (Alfred Stutevant).  

Armed with the knowledge of the location of the 10 genes that contributed to the so-called bithorax complex (due to Sturtevant's mapping technology), Lewis around 1960 created a model of how the 10 bithorax HOX genes evolved from an ancestral gene through tandem gene duplications (which is largely thought correct) which culminated in his theoretical paper in the late 1970s.  Lewis knew through his own genetic gain and loss of function assays (due to classical genetics Lewis could create hybrid flies due to recombiantion during meiosis to create crosses of known mutant HOX lineages of flies, such as Thomas's line).  From these experiments he constructed a Wolpert-like gradient model of how the HOX genes would interact like a Fourier series, one HOX gene would be coarsly expressed, thereby turning on another HOX gene in the complex, then the two of these would work in tandem to turn on a third HOX gene, then the three would all work in unison to turn on the fourth HOX gene.  Central to his hypothesis was not just the consecutive combinations of the HOX gene products activate new HOX genes in the complex, but also that each new actively generated HOX gene would repress the most recent in time activated gene, thereby creating fine patterns of gene expressions within the cells of the embryo.  These unique gene products across the embryo were isolated in segments in space (he could literally see the segments, as these were gross anatomical features each containing 1000's of cells in early development).  

Hence, Lewis had proposed a developmental mechanism for how each segment contained different sets of gene products, and thereby explained how each segment would set in motion the signalling and paracrine factors that would eventually cause the segment to form an anatomical feature like a leg, wing, head, or tail.  The HOX genes are transcription factors (master genes) that target the genes necessary in signalling pathways and paracrine induction.   In short, Lewis had a hypothesis for how anatomical features were built from the segmented larva, but more importantly, he saw an evolutionary mechanism for saltation or macroevolution by his model of the HOX genes; as the basis of HOX genes is that if you mutate them, then the segments of the fly would change in such a way to suggest that the anatomical features that decorate the segments (antennea, wings, legs, eye, abdomen etc..) were all just one ancestral anatomical feature like a leg for locomotion, that could be adapted or modified for further purposes, like an antennae for sensing the environment, or a mandible or claw for killing prey.

The question of how the segmented larva developed from the fertilized egg was being elucidated in parallel with Lewis' work, through experiments on oogenesis and early devolopment.  Partly through the extensive genetic mutation experiments of Nullsein Volhard and Eric Wieschous it became clearer that there were maternal laid and zygotic genes (transcription factors) that caused segmentation in the larva from the humble beginning of the fertilized egg, these genes are known as the gap genes and the pair-rule genes due to the mutant patterns that the would cause in the early embryo and larva.  

Although a genetic description was emerging of how the fly develops, it was not possible to isolate the gene products at that time due to the gene's protein being inside of a multicellular environment.  Lewis knew the locations on the chromosome of what gene's caused homeotic mutations, and what segments of a larva were effected by the mutation; but Lewis simply didn't know to what extent that gene's product caused the mutation (it obviously could have been through some convoluted web of interactions).  Work by McGinnis and Levine in Gerhart's lab, in the early 1980s, possibly motivated by Lewis's hypothesis that \textbf{all} the HOX genes are paralogs, lead to the development of a molecular optical microscopic staining technique that allowed one to visually confirm and isolate the region of expression of each HOX gene in specific segments of the fly.  As Lewis had hypothesized, every HOX gene was identical (they were paraloges), all of which contained a DNA binding domain called the homeobox, a stretch of about 100 amino acids that encodes the binding domain.

A combination of the new microscopic technique, along with pattern recognition techniques has now allowed for developmental biology to quickly advance by elucidating each gene expressed in each cell, and if that gene's product is a transcription factor, where that gene binds within the genome to determine the targets of the factor.  


\subsection{Multiple Sequence Alignment}
Conservation of genetic elements (conserved DNA sequences) is the basis of similarities in Linnean groups, in species.  Mutation of genetic elements is the basis of difference in species.  Conservation of genetic elements is as simple as comparing to words (sequences) by checking if they 'match' at each position.


 



  
\subsection{Conserved Gene Regulatory Networks}